src/RestApi.php

```diff
@@ -61,19 +61,19 @@ public static function get_routes_from_namespace( $namespace ) {
 	 *
 	 * @todo check compat < WC 3.6. Make specific to cart endpoint.
 	 * @param mixed $return Value being filtered.
-	 * @param array $request Request data.
 	 * @return mixed
 	 */
-	public static function maybe_init_cart_session( $return, $request = false ) {
-		// Pass through other errors.
-		if ( ! empty( $error ) ) {
-			return $error;
+	public static function maybe_init_cart_session( $return ) {
+		$wc_instance = wc();
```

Comment:
I see what you want to do here (protect against `wc()` returning no instance), however, what scenario is there where this becomes a problem?

The `wc()` function always returns an instance (seen [here](https://github.com/woocommerce/woocommerce/blob/master/woocommerce.php#L51-L53)) so this check might not be needed?

If we did want to keep a check, it might be even more safe to check for the existence (is_callable) for `wc` so it's not attempted to be used when unavailable?

Reply:
Ahh, so I probably wrongly assumed the fatal error reported in #1697 meant `wc` was not returning an instance WC when instead, it's likely that the `WooCommerce::initialize_session` method is not present in whatever environment this user has Woo Blocks installed in (which I see was added in WC 3.6.4).  I thought blocks would be prevented from being activated if the minimum required version of WooCommerce is not active (3.7?) but it looks like that might not be the case?

Either way, I agree that checking for an instance of `wc()` is not the issue needing fixed, instead I can check if the `initialzie_session` method is available.